src/composables/graph/useGraphNodeManager.ts

```diff
@@ -66,6 +66,17 @@ export function useGraphNodeManager(graph: LGraph): GraphNodeManager {
   // Non-reactive storage for original LiteGraph nodes
   const nodeRefs = new Map<string, LGraphNode>()
 
+  const syncNodeSlotData = (nodeId: string, node: LGraphNode) => {
+    const currentData = vueNodeData.get(nodeId)
+    if (!currentData) return
+
+    vueNodeData.set(nodeId, {
+      ...currentData,
+      inputs: node.inputs ? [...node.inputs] : undefined,
```

Comment:
Not for this PR, but what if we defined inputs and outputs as being required, so that a lack of them would just be an empty array? Would that simplify some of the logic?